dominant-strategies/quaipay
# feat: add receive amount input screen

## Description

Continuing on the flow to receive funds, the amount input screen is where the user can type in the specific amount and then follow-up with the generated QR code.

On this PR, we added the basic functionality where the user can input the desired amount both in USD and in QUAI (exchange rate to be determined — hardcoded for now).

Next steps for this screen are:
- Improve its Look & Feel to match the design
- Get dynamic exchange rate
- Handle input's edge cases with proper logic for the keyboard + input

## Related links

- #44 

## Demo

| _iOS (iPhone SE 2020)_ | _Android (720x1200 w/ Pie 9)_ |
| :---: | :---: |
| <video src='https://github.com/dominant-strategies/quaipay/assets/21087992/2640715b-1f71-42b3-9b59-b53e06266fb7' width=400> | <video src='https://github.com/dominant-strategies/quaipay/assets/21087992/cce48f18-2103-4783-8ac6-1e5a959ec73e' width=400> |

## Testing instructions

- Open the app with an onboarded user (if not, follow onboarding instructions)
- On the Home Screen tab, on Receive in the top bar, press on "Request"
  - Check that the user browses to the receive amount input screen
    - _Back navigation_
      - Check that on back press in the top left corner, the user goes back to the home screen
      - Check that on native back navigation (swipe from left corner for iOS  — physical back button for Android), the user goes back
    - _Swap button_
      - Check that on press of the swap button, amounts are visually swapped
      - Check that on swap, new amount can be edited via input 
    - _Input_
      - Check that native keyboard is prompted on screen mount (i.e. focus on input is made on screen mount)
      - Check that on first number input, the initial value is replaced by the number.
      - Check that on decimal comma input, the initial value and any other previous value is kept before it
        - [BUG] Multiple commas could be appended resulting in an invalid number for now
      - Check that on full deletion for the previous value, the input is reset to the initial amount
      - Check that on input change, the equivalent is correctly updated (by hardcoded exchange rate value)
